.github/agents/meshkit-code-contributor.md

```diff
@@ -0,0 +1,199 @@
+
+## MeshKit Purpose and Responsibilities
+
+MeshKit provides shared **error handling** and related tooling across Meshery’s Go services. It centralizes error code definitions, enriches errors with context, and offers utilities to generate and analyze error code metadata.
+
+### Error Handling Patterns
+
+- Use MeshKit’s `errors` package for all new error definitions.  
+- Assign stable error codes and use structured metadata to provide alerts, causes, remedies, and references.
+- Keep error messages actionable and consistent across Meshery components.
+
+### Errorutil Utility
+
+MeshKit includes an `errorutil` command for working with error codes:
+
+- **Update mode** – Scans the codebase and generates or updates error code definitions:  
+  - `make errorutil`  
+  - Under the hood: `go run github.com/meshery/meshkit/cmd/errorutil -d . update --skip-dirs meshery -i .helpers -o .helpers`  
+- **Analyze mode** – Analyzes existing error codes for consistency and gaps:  
+  - `make errorutil-analyze`  
+  - Uses the same module with the `analyze` subcommand.  
+- **Build binary** – `make build-errorutil` builds an `errorutil` binary for direct use.
+
+## Core Competencies
+
+1. **MeshKit Error Model**  
+- Define errors using MeshKit’s `errors` package: codes, severity, user-facing messages, causes, and remediation steps.
+- Maintain uniqueness and stability of error codes across the library.
+
+2. **Shared Utility Design**  
+- Implement reusable helpers that can be consumed safely by Meshery components.  
+- Avoid coupling MeshKit directly to specific services; keep APIs generic and well-documented.
+
+3. **Testing and Validation**  
+- Write unit tests using Go’s standard testing library, targeting short, deterministic tests.
+- Use `make test` to execute tests with race detection and coverage.
+
+4. **Build System Proficiency**  
+- Use MeshKit’s Makefile targets instead of raw `go` commands when equivalents exist.
+- Treat failures in `make test`, `make check`, or `make tidy` as issues to fix rather than bypass.
+
+5. **Errorutil Workflows**  
+- Run `make errorutil` after introducing new errors to regenerate error code artifacts.
+- Use `make errorutil-analyze` to validate the consistency and health of error code definitions.
+
+### Preferred Workflow
+- Use `make test` for running tests (`--short`, `-race`, coverage) instead of direct `go test`.
+- Run `make check` for linting and fix all reported issues before opening a PR.
+- Run `make tidy` to ensure dependency metadata is clean and fails if module files change unexpectedly.
+- When adding or modifying error definitions, run `make errorutil` and, when needed, `make errorutil-analyze`.
+
+## Code Style and Conventions
+
+### Go Code Standards
+
+```go
+// Follow standard Go conventions and formatting (gofmt, goimports).
+// Use golangci-lint with the repository's .golangci.yml configuration.
+
+// Example: MeshKit-style error definition (pattern adapted from Meshery spec)
```

Comment:
Good to see examples included. This helps set the record straight for the agent and has them fall inline with our existing conventions.